aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -387,6 +393,36 @@ std::tuple<Tensor, Tensor> native_multi_head_attention(
   const auto dim_per_head = D / num_head;
 #endif
 
+  const int64_t sdp_dim_per_head = D / num_head;
```

Comment:
I am 98% this is not going to pass CI because sdp_utils uses some cuda specific functions to check the device arch. I tried using #ifdef USE_CUDA but for some reason this was not working so for now I added

Reply:
nit: use `dim_per_head` from above